feat: when allowMetadataInput is enabled, accept created and user fields

bkoski opened the pull request:
The backend APIs work nicely when importing content.  The only hitch is that you can't import content with a creation time or attribution, so all Q&A entries appear as if they were created today by the user who ran the load.

This PR adds an `allowMetadataInput` config flag that, when enabled, allows `created` and `author` fields to be passed along in the POST request to override the default handling.  Normally this should be flagged false except during initial import or loading.

One major area of change is that previously the DatabaseQetaStore always used the current time as the `created` value. This PR requires that a `created` value is explicitly passed into persistence functions.  This PR also adds a `getCreated` function to the utils in parallel to `getUsername` that either accepts these fields from input or uses the current date as a default.

Curious to get your thoughts!

drodil reviewed (approved):
Looks good this far. How about comments? I think that was not supported yet?

bkoski commented:
Thanks! Since this change is in utils, good news is it works equally for all routes including comments.  But comments weren't tested; just updated PR with tests for both question and answer comments.  Also fixed tsc errors.
